freedomofpress/securedrop
# Remove runtime javascript minification

## Status

Ready for review

## Description of Changes

Fixes #6415

- serves static/js resources instead of gen/js resources
- removes flask-assets, jsmin, and webassets dependencies and related code
- updates apparmor profile to remove rw access to /gen/
- updates builder functionality/tests to ensure gen/ not present
- updates postinst to remove gen/, .webassets-cache/ if present

## Testing

- checkout this branch
- run `git clean -dfx` to ensure old generated files aren't left hanging around
- run `make dev-tor`
  - [x] confirm that applications start successfully
  - [x] confirm that `source.js` on the SI home page is served from `/static/js/`, not `/static/gen`
  - [x] visit the SI via a tor2web proxy and verify that you are immediately redirected to the `/tor2web-warning` page on visiting the SI home page
  - [x] log into the JI and confirm that `journalst.js` on the JI All Source page is served from `/static/js/`, not `/static/gen`
  - [x] confirm that the source filter still works
- [x] run `make build-debs`, confirm that the package build completes without error
- [x] run `make staging`, and confirm that staging servers are provisioned without error and that `source.js` and `journalist.js` are served from `/static/js` on the   SI and JI respectively

### Upgrade testing (optional)
- [ ] run through the upgrade scenario with this branch and confirm that the upgrade completes without error and that `source.js` and `journalist.js` are served from `/static/js` on the   SI and JI respectively

## Deployment

This  PR modifies the `securedrop-app-code` `postint` to remove the `/static/gen` and `/static/.webassets-cache` directories in the webroot, so upgrade testing should be performed prior to release.

## Checklist

### If you made changes to the server application code:


### If you made changes to `securedrop-admin`:


### If you added or removed a file deployed with the application:


### If you made non-trivial code changes:


Choose one of the following:
